File: api/app/schemas/leagues.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class LeagueCreate(BaseModel):
    name: str
    slug: str
    description: str | None = Field(default=None, max_length=1000)


class LeagueUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    description: str | None = Field(default=None, max_length=1000)


class LeagueRead(BaseModel):
    id: UUID
    name: str
    slug: str
    plan: str
    driver_limit: int
    owner_id: UUID | None
    is_deleted: bool
    deleted_at: datetime | None
    description: str | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/leagues.ts
import type { LeagueRole } from './auth'

export interface LeagueRead {
  id: string
  name: string
  slug: string
  plan?: string | null
  driver_limit?: number | null
  owner_id?: string | null
  is_deleted?: boolean
  deleted_at?: string | null
}

export interface LeagueSummary {
  id: string
  name: string
  slug: string
  plan: string | null
  driverLimit: number | null
  role: LeagueRole | null
}

export interface CreateLeagueRequest {
  name: string
  slug: string
}

export interface LeagueEventSummary {
  id: string
  name: string
  track: string
  startTime: string
  status: string
}

export interface LeagueResultPodiumEntry {
  driverName: string
  teamName?: string | null
  position: number
  points?: number | null
}

export interface LeagueResultSummary {
  event: LeagueEventSummary
  podium: LeagueResultPodiumEntry[]
}

export interface LeagueOverviewData {
  league: LeagueSummary & {
    description?: string | null
  }
  nextEvent?: LeagueEventSummary | null
  recentResult?: LeagueResultSummary | null
  discordLinked: boolean
}
